mmcv/runner/epoch_based_runner.py

```diff
@@ -90,19 +91,25 @@ def run(self, data_loaders, workflow, max_epochs, **kwargs):
         assert len(data_loaders) == len(workflow)
 
         self._max_epochs = max_epochs
+
+        for i, flow in enumerate(workflow):
+            mode, epochs = flow
+            if mode == 'train':
+                self._max_iters = self._max_epochs * len(data_loaders[i])
+                break
```

Comment:
Can we explain the necessity here? For example, which hook's `before_run` function requires the self._max_iters to be calculated.

Reply:
Never mind, I have rewind the change of order back. In some hooks such as `MomentumUpdaterHook`, self._max_iters is necessary, and the previous order changing will cause them to fail. I will not change the order now, and update the self._max_iters in my own hook.